pkg/segment/query/processor/queryprocessor.go

```diff
@@ -58,13 +58,21 @@ func NewQueryProcessor(firstAgg *structs.QueryAggregators, queryInfo *query.Quer
 		return nil, utils.TeeErrorf("NewQueryProcessor: cannot make searcher; err=%v", err)
 	}
 
+	_, queryType := query.GetNodeAndQueryTypes(&structs.SearchNode{}, firstAgg)
+
 	dataProcessors := make([]*DataProcessor, 0)
 	for curAgg := firstAgg; curAgg != nil; curAgg = curAgg.Next {
+		// If query Type is GroupByCmd/SegmentStatsCmd, this agg will be processed by the searcher.
+		if queryType != structs.RRCCmd {
+			continue
+		}
```

Comment:
So if we get to the `continue` we expect the `curAgg` to be a stats command right? I think it would be good to also check that before doing `continue`; that way:
1. It's easier to debug if we ever `continue` when we shouldn't have
2. It makes the code more readable

Reply:
Addressed in the latest commit.